File: t/VB/DicomServerDemo/Database/DBBase.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Data

Namespace Leadtools.Demos.Database
   ''' <summary>
   ''' Summary description for DBBase.
   ''' </summary>
   Public Class DBBase
      Protected ds As DataSet = New DataSet("DICOMSVR")
      Protected dbFileName As String

      Public ReadOnly Property Count() As Integer
         Get
            Return ds.Tables(0).Rows.Count
         End Get
      End Property

      Public ReadOnly Property DB() As DataSet
         Get
            Return ds
         End Get
      End Property

      Public Sub New()
         '
         ' TODO: Add constructor logic here
         '
      End Sub

      Protected Function LoadDatabase() As Boolean
         Dim loaded As Boolean = True

         Try
            ds.ReadXml(dbFileName)
         Catch e As Exception
            loaded = False
         End Try
         Return loaded
      End Function

      Public Function Save() As Boolean
         Dim ret As Boolean = False

         Try
            ds.WriteXml(dbFileName, XmlWriteMode.WriteSchema)
            ret = True
         Catch
         End Try
         Return ret
      End Function

   End Class
End Namespace

File: t/CS/DicomServerDemo/Database/DBBase.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Data;

namespace Leadtools.Demos.Database
{
   /// <summary>
   /// Summary description for DBBase.
   /// </summary>
   public class DBBase
   {
      protected DataSet ds = new DataSet("DICOMSVR");
      protected string dbFileName;

      public int Count
      {
         get
         {
            return ds.Tables[0].Rows.Count;
         }
      }

      public DataSet DB
      {
         get
         {
            return ds;
         }
      }

      public DBBase( )
      {
         //
         // TODO: Add constructor logic here
         //
      }

      protected bool LoadDatabase( )
      {
         bool loaded = true;

         try
         {
            ds.ReadXml(dbFileName);
         }
         catch
         {
            loaded = false;
         }
         return loaded;
      }

      public bool Save( )
      {
         bool ret = false;

         try
         {
            ds.WriteXml(dbFileName, XmlWriteMode.WriteSchema);
            ret = true;
         }
         catch
         {
         }
         return ret;
      }
   }
}
